### PR TITLE
Resolved linagora/Twake#1467

### PR DESCRIPTION
**Pull request by romanesko**
company admin is now always a workspace admin for his company

### CODE REVIEW

File: twake/backend/node/src/utils/workspace.ts [hunk 1]
```diff
@@ -13,6 +13,14 @@ export async function isWorkspaceAdmin(
     return false;
   }
 
+  const companyUser = await userService.companies.getCompanyUser(
+    { id: workspace.company_id },
+    { id: user.id },
+  );
+  if (companyUser && companyUser.role == "admin") {
```
Comment (RomaricMourgues) [inline twake/backend/node/src/utils/workspace.ts:20]: owner also have this right

File: twake/backend/node/src/services/workspaces/services/workspace/service.ts [hunk 2]
```diff
@@ -205,6 +205,7 @@ export class WorkspaceService implements WorkspaceServiceAPI {
     companyId: CompanyPrimaryKey,
   ): Promise<WorkspaceUser[]> {
     const allCompanyWorkspaces = await this.getAllForCompany(companyId.id);
+    // const userCompany: UserCompany = await this.company
```
Comment (RomaricMourgues) [inline twake/backend/node/src/services/workspaces/services/workspace/service.ts:208]: Remove this comment

File: twake/backend/node/src/services/workspaces/web/controllers/workspaces.ts [hunk 3]
```diff
@@ -106,9 +106,17 @@ export class WorkspacesCrudController
...
+      { id: context.company_id },
+      { id: context.user.id },
+    );
+
     const allUserWorkspaceRolesMap = await this.workspaceService
       .getAllForUser({ userId: context.user.id }, { id: context.company_id })
-      .then(uws => new Map(uws.map(uw => [uw.workspaceId, uw.role])));
+      .then(
+        uws =>
+          new Map(uws.map(uw => [uw.workspaceId, companyUser.role == "admin" ? "admin" : uw.role])),
```
Comment (RomaricMourgues) [inline twake/backend/node/src/services/workspaces/web/controllers/workspaces.ts:118]: company "owner" too is admin of the workspace

**Review by RomaricMourgues — CHANGES_REQUESTED** [review]
Small changes needed.

We should set the "admin" override on the "get" method too (on the top of the "list" method)

File: twake/backend/node/src/utils/workspace.ts (hunk 1 above)
Comment (RomaricMourgues) [reply twake/backend/node/src/utils/workspace.ts:20]: Now you have utils functions to get if this is a "at least" admin level. Search for `hasCompanyAdminLevel` in project

File: twake/backend/node/src/services/workspaces/web/controllers/workspaces.ts (hunk 3 above)
Comment (RomaricMourgues) [reply twake/backend/node/src/services/workspaces/web/controllers/workspaces.ts:118]: Now you have utils functions to get if this is a "at least" admin level. Search for `hasCompanyAdminLevel` in project

Comment (romanesko) [reply twake/backend/node/src/services/workspaces/web/controllers/workspaces.ts]: fixed

**Review by chamerling — APPROVED** [review]

**Pull request merged**